### ISSUE TITLE
Refactor UI for usability with Mantine

### ISSUE THREAD

**Issue by nkanchinadam** [Hack4Impact-UMD/swaliga-foundation#49]
Currently, many of the UI components are not standardized and lack basic accessibility features. Refactor the site's UI to use Mantine to take advantage of its global theming and pre-built flexible components.
